examples/plugin/reset_plugin/joint_position_randomizer.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/math/Rand.hh>
+#include <ignition/gazebo/System.hh>
+#include <ignition/gazebo/components/Joint.hh>
+#include <ignition/gazebo/components/JointPositionReset.hh>
+#include <ignition/gazebo/components/ParentEntity.hh>
+#include <ignition/plugin/Register.hh>
+
+using namespace ignition;
+using namespace gazebo;
+namespace reset_plugin
+{
+class JointPositionRandomizer : public System,
+                                public ISystemConfigure,
+                                public ISystemReset
+{
+  void Configure(const Entity &_entity,
+                 const std::shared_ptr<const sdf::Element> &,
+                 EntityComponentManager &, EventManager &) override
+  {
+    this->targetEntity = _entity;
+  }
+
+  void Reset(const UpdateInfo &_info, EntityComponentManager &_ecm) override
+  {
+    std::cout << "Reset joints\n";
+    auto joints = _ecm.EntitiesByComponents(
+        components::ParentEntity(this->targetEntity), components::Joint());
+    for (auto joint : joints)
+    {
+      double pos = math::Rand::DblUniform(0, IGN_PI);
```

Comment:
We probably want a different range for the prismatic joint

![resetplugin](https://user-images.githubusercontent.com/5751272/162548432-1e46bde5-808a-4e4e-a751-0c38f8143193.gif)

Reply:
Done in [3799ff3](https://github.com/ignitionrobotics/ign-gazebo/pull/1327/commits/3799ff3f5e4030e0f0351c56b05fa6a06295fd87)